sql/catalog/lease: TestNameCacheEntryDoesntReturnExpiredLease failed

sql/catalog/lease.TestNameCacheEntryDoesntReturnExpiredLease [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14609847?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/14609847?buildTab=artifacts#/) on master @ [a3ac7ebf958f25201c2696a17f996c0b9f86830f](https://github.com/cockroachdb/cockroach/commits/a3ac7ebf958f25201c2696a17f996c0b9f86830f):


```
=== RUN   TestNameCacheEntryDoesntReturnExpiredLease
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/be0d053872b35521d816dbd3d2572df0/logTestNameCacheEntryDoesntReturnExpiredLease2099848132
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_internal_test.go:521: name cache has unexpired entry for (104, test): 106("test",010180c8d491ed985a47ffa12e1298d2066344) ver=1:1711669634.232849041,0, refcount=1
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/be0d053872b35521d816dbd3d2572df0/logTestNameCacheEntryDoesntReturnExpiredLease2099848132
--- FAIL: TestNameCacheEntryDoesntReturnExpiredLease (0.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestNameCacheEntryDoesntReturnExpiredLease.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37181

--- a/pkg/sql/catalog/lease/lease_internal_test.go
+++ b/pkg/sql/catalog/lease/lease_internal_test.go
@@ -487,6 +487,11 @@ CREATE TABLE t.%s (k CHAR PRIMARY KEY, v CHAR);
 		t.Fatal(err)
 	}
 
+	// Disable stats collection so that the descriptor isn't modified.
+	if _, err := db.Exec("SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false"); err != nil {
+		t.Fatal(err)
+	}
+
 	// Populate the name cache.
 	if _, err := db.Exec("SELECT * FROM t.test;"); err != nil {
 		t.Fatal(err)

